HelmChartInflationGeneratorPlugin: newer helm versions omit `v` in returned string

```
# helm3 version -c --short                                        
v3.1.0+gb29d20b

# helm version -c --short
3.5.2+gc4e7485
```

I assume this will break the check on 
https://github.com/kubernetes-sigs/kustomize/blob/a1df3e030ff29e5192aa7f5d8ed2cf4291630ec6/api/builtins/HelmChartInflationGenerator.go#L267

Locally i get:

```
{Version:kustomize/v3.8.9 GitCommit:f30fea4c072c794e9b936f20bd1592d9fa25d976 BuildDate:2020-12-29T15:49:08Z GoOs:linux GoArch:amd64}

# kustomize build --stack-trace xxx/dev/us-central1
panic: runtime error: slice bounds out of range [1:0]

goroutine 1 [running]:
sigs.k8s.io/kustomize/api/builtins.(*HelmChartInflationGeneratorPlugin).checkHelmVersion(0xc0009ce360, 0x413d4a, 0xc002689dd0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/builtins/HelmChartInflationGenerator.go:241 +0x2f6
sigs.k8s.io/kustomize/api/builtins.(*HelmChartInflationGeneratorPlugin).Generate(0xc0009ce360, 0x0, 0x0, 0x0, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/builtins/HelmChartInflationGenerator.go:159 +0x7a
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).runGenerators(0xc000311a40, 0xc000482b40, 0xc0025b6a80, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:224 +0x1b3
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateTarget(0xc000311a40, 0xc000482b40, 0x40, 0x0, 0x165dc01)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:190 +0x255
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).AccumulateTarget(0xc000311a40, 0x0, 0x0, 0x198dac0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:155 +0xc6
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateDirectory(0xc000311280, 0xc0004824e0, 0x19755c0, 0xc000548c60, 0xc000548c00, 0x0, 0x0, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:401 +0x695
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateResources(0xc000311280, 0xc0004824e0, 0xc000311940, 0x4, 0x4, 0x7f030f64e108, 0x0, 0xc0005994a8)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:345 +0x187
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateTarget(0xc000311280, 0xc0004824e0, 0x40, 0x0, 0x165dc01)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:162 +0x74
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).AccumulateTarget(0xc000311280, 0x0, 0x0, 0x198dac0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:155 +0xc6
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateDirectory(0xc000311000, 0xc0004820e0, 0x19755c0, 0xc000548840, 0xc000548800, 0x0, 0x0, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:401 +0x695
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateResources(0xc000311000, 0xc0004820e0, 0xc000311040, 0x2, 0x4, 0x7f030f64e108, 0x0, 0xc000599780)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:345 +0x187
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateTarget(0xc000311000, 0xc0004820e0, 0x40, 0x0, 0x165dc01)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:162 +0x74
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).AccumulateTarget(0xc000311000, 0x0, 0x0, 0x198dac0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:155 +0xc6
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateDirectory(0xc00003abc0, 0xc0003cd8a0, 0x19755c0, 0xc000548420, 0xc000548400, 0x0, 0x0, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:401 +0x695
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateResources(0xc00003abc0, 0xc0003cd8a0, 0xc000310ec0, 0x1, 0x4, 0x7f030f64e108, 0x0, 0xc000599a58)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:345 +0x187
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateTarget(0xc00003abc0, 0xc0003cd8a0, 0xc000000180, 0x4, 0x1)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:162 +0x74
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).AccumulateTarget(0xc00003abc0, 0x40, 0x0, 0x190fbb2)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:155 +0xc6
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).makeCustomizedResMap(0xc00003abc0, 0x0, 0x0, 0x198dac0, 0x25c8208)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:110 +0x2f
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).MakeCustomizedResMap(...)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/internal/target/kusttarget.go:106
sigs.k8s.io/kustomize/api/krusty.(*Kustomizer).Run(0xc0012d5d20, 0x7ffe51f6212c, 0x22, 0x0, 0x0, 0x0, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.6.8/krusty/kustomizer.go:77 +0x294
sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build.(*Options).RunBuild(0xc000467260, 0x1934a40, 0xc000182008, 0x2, 0x0)
	/workspace/myClone/kustomize/internal/commands/build/build.go:148 +0xb5
sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build.NewCmdBuild.func1(0xc0002d6840, 0xc0003cc660, 0x1, 0x2, 0x0, 0x0)
	/workspace/myClone/kustomize/internal/commands/build/build.go:70 +0xab
github.com/spf13/cobra.(*Command).execute(0xc0002d6840, 0xc0003cc620, 0x2, 0x2, 0xc0002d6840, 0xc0003cc620)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002d62c0, 0x0, 0x14dad40, 0xc000096058)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	/workspace/myClone/kustomize/main.go:19 +0x27
```

--- a/api/builtins/HelmChartInflationGenerator.go
+++ b/api/builtins/HelmChartInflationGenerator.go
@@ -270,11 +270,17 @@ func (p *HelmChartInflationGeneratorPlugin) checkHelmVersion() error {
 	if err != nil {
 		return err
 	}
-	r, err := regexp.Compile(`v\d+(\.\d+)+`)
+	r, err := regexp.Compile(`v?\d+(\.\d+)+`)
 	if err != nil {
 		return err
 	}
-	v := string(r.Find(stdout))[1:]
+	v := r.FindString(string(stdout))
+	if v == "" {
+		return fmt.Errorf("cannot find version string in %s", string(stdout))
+	}
+	if v[0] == 'v' {
+		v = v[1:]
+	}
 	majorVersion := strings.Split(v, ".")[0]
 	if majorVersion != "3" {
 		return fmt.Errorf("this plugin requires helm V3 but got v%s", v)

--- a/plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go
+++ b/plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go
@@ -278,11 +278,17 @@ func (p *HelmChartInflationGeneratorPlugin) checkHelmVersion() error {
 	if err != nil {
 		return err
 	}
-	r, err := regexp.Compile(`v\d+(\.\d+)+`)
+	r, err := regexp.Compile(`v?\d+(\.\d+)+`)
 	if err != nil {
 		return err
 	}
-	v := string(r.Find(stdout))[1:]
+	v := r.FindString(string(stdout))
+	if v == "" {
+		return fmt.Errorf("cannot find version string in %s", string(stdout))
+	}
+	if v[0] == 'v' {
+		v = v[1:]
+	}
 	majorVersion := strings.Split(v, ".")[0]
 	if majorVersion != "3" {
 		return fmt.Errorf("this plugin requires helm V3 but got v%s", v)

